src/dataset-builder/DatasetBuilder.ts

```diff
@@ -744,21 +752,22 @@ export const DatasetBuilderView: React.FC<DatasetBuilderProps> = (props) => {
                 return h(DomainCriteriaSelector, {
                   state: datasetBuilderState,
                   onStateChange,
-                  datasetId,
+                  snapshotId,
                   getNextCriteriaIndex,
                 });
               case 'domain-criteria-search':
                 return h(DomainCriteriaSearch, {
                   state: datasetBuilderState,
                   onStateChange,
-                  datasetId,
+                  snapshotId,
                   getNextCriteriaIndex,
                 });
               case 'concept-set-creator':
-                return datasetModel.state.snapshotBuilderSettings
+                return snapshotBuilderSettings.status === 'Ready'
```

Comment:
Same thing, I don't think you need this check because it is tested above before render